components/aws/sagemaker/common/_utils.py

```diff
@@ -471,10 +471,52 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  try:
+      endpoint_name = client.describe_endpoint(EndpointName=endpoint_name)['EndpointName']
+      logging.info("Endpoint exists: " + endpoint_name)
+      return True
+  except ClientError as e:
+      logging.info("Endpoint does not exist")
```

Comment:
I was going to comment that using `try/catch` logic as conditions was an anti-pattern, but [this stack overflow post](https://stackoverflow.com/questions/16138232/is-it-a-good-practice-to-use-try-except-else-in-python) has proved to me that Python is one of the few languages where it is alright. How strange.

I might add that can we change `logging.info` to `logging.debug` for these statements - I don't think printing these statements would be useful for the user.

Reply:
I wasn't aware of this for Python, thanks for the pointer. I changed the logging to debug